rust/lancedb/src/io/object_store.rs

```diff
@@ -156,15 +188,15 @@ impl ObjectStore for MirroringObjectStore {
         if to.primary_only() {
             self.primary.copy(from, to).await
         } else {
-            self.secondary.copy(from, to).await?;
+            self.secondary_copy(from, to).await?;
             self.primary.copy(from, to).await?;
             Ok(())
         }
     }
 
     async fn copy_if_not_exists(&self, from: &Path, to: &Path) -> Result<()> {
         if !to.primary_only() {
-            self.secondary.copy(from, to).await?;
+            self.secondary_copy(from, to).await?;
```

Comment:
could we just do a hard link? and soft fail?

Reply:
also fyi the local fs object store does do a hard link when copying:
https://github.com/apache/arrow-rs/blob/object_store_0.9.0/object_store/src/local.rs#L601-L616